File: trees_graphs/closest_value/go/closest_value.go
package closest_value

import "math"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

const (
	bigFloat = float64(1 << 64)
)

func ClosestValue(root *TreeNode, target float64) int {
	closest := -1
	minDistance := bigFloat

	el := root
	for el != nil {
		distance := math.Abs(float64(el.Val) - target)
		if distance < minDistance {
			closest = el.Val
			minDistance = distance
		}

		if float64(el.Val) < target {
			el = el.Right
		} else {
			el = el.Left
		}
	}

	return closest
}

File: trees_graphs/closest_value/python/closest_value.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def closest_value(self, root: TreeNode, target: float) -> int:
        closest: int = -1
        minDifference = float("inf")
            
        el = root
        while el != None:
            diff = abs(el.val - target)
            if diff < minDifference:
                minDifference = diff
                closest = el.val
            
            if el.val < target:
                el = el.right
            else:
                el = el.left
                
        return closest